tokensRemover/metaDataRemover/txCreator.go

```diff
@@ -75,22 +80,39 @@ func createShardTxs(
 	return nil
 }
 
-func createTxs(
+type txCreator struct {
+	proxy        proxyProvider
+	txInteractor transactionInteractor
+}
+
+func newTxCreator(proxy proxyProvider, txInteractor transactionInteractor) (*txCreator, error) {
+	if proxy == nil {
```

Comment:
why not add `IsInterfaceNil() bool` to the interfaces and stub and replace with `check.IfNil(proxy)`, `check.IfNil(txInteractor)`

Reply:
IMHO, there is no need to "over-complicate" some simple unexported interfaces only used for a script by adding extra func and extend it for stubs for this single line of code

Reply:
at least please use check.IsNilReflect as in some situation this check does not work as expected
Or, being an unexported sub-component you can drop the checks at all, including this constructor

Reply:
As discussed with @iulianpascalau , we've decided to completely remove the constructor, since it is unexported